### ISSUE TITLE
Rust debugging with lldb-vscode

### ISSUE THREAD

**Issue by LuigiPiucco** [emacs-lsp/dap-mode#366]
### Context
I'm trying to debug a Rust program with lldb though [lldb-vscode](https://github.com/llvm/llvm-project/tree/master/lldb/tools/lldb-vscode). It doesn't seem to be supported out of the box from looking at [dap-lldb.el](https://github.com/emacs-lsp/dap-mode/blob/master/dap-lldb.el), as it hardcodes the VSCode extension bundled version, while I have a system install. I tried to use `dap-register-debug-provider` and `dap-register-debug-template` to do it, but it doesn't seem to be working, as detailed.

### Details
TL;DR this is the configuration that got me closest (I think):
```elisp
(defun get-rust-binary ()
  "Prompt user for which binary to use and return the filename."
  (interactive)
  (let* ((bin-name (read-string "Enter the target to execute: "))
         (filename (concat (projectile-project-root) "target/debug/" bin-name))
         (does-not-exist (not (file-exists-p filename))))
    (when does-not-exist (message "No binary found for especified target %s!" bin-name))
    filename))
(defun get-rust-args ()
  "Prompt user for args to pass."
  (interactive)
  (let ((args-string (read-string "Enter the target arguments: ")))
     (split-string args-string)))
(defun dap-lldb-rust--populate-start-file-args (conf)
  "Populate CONF with the required arguments."
  (-> conf
      (dap--put-if-absent :dap-server-path '("/usr/bin/lldb-vscode" "1441"))
      (dap--put-if-absent :port 1441)
      (dap--put-if-absent :type "lldb")
      (dap--put-if-absent :cwd (car `(,(projectile-project-root))))
      (dap--put-if-absent :program (car `(,(get-rust-binary))))
      (dap--put-if-absent :name "Rust Debug")
      (dap--put-if-absent :program-to-start "cargo build")
      (dap--put-if-absent :args (car `(,(get-rust-args))))))
(add-hook 'rustic-mode-hook (lambda ()
                              (dap-register-debug-provider "lldb" 'dap-lldb-rust--populate-start-file-args)
                              (dap-register-debug-template "Rust Run Configuration"
                                                           `(:type "lldb"
                                                             :cwd ,(projectile-project-root)
                                                             :request "launch"))))
```
What it's trying to do is create a provider and template where the user (me, in this case) is prompted for the name of the executable to debug and what arguments to pass to it. The prompts are done in `get-rust-binary` and `get-rust-args`, which are working, I get prompted and no error happens if I input an existing executable. I also tried hardcoding the paths temporarily, I get the same results. The compilation step in `:program-to-start`is also working.

Something to note is the port 1441 passed to `:port` and as an arg to lldb-vscode, I had problems with #346 if I used a single element list for `:dap-server-path`, so I passed a port to it. Even if I pass an empty string and no `:port`, the same happens.

What's failing is I don't get any output after all that. A window opens with the out buffer for the debug session, but there's nothing in it, and I get "Debug Adapter finished" immediately. On looking at my process manager, lldb-vscode is running, but I see no sign of my program.

If you need more information than what I initially I thought relevant, or if I can contribute in any way, do tell me, I willing to help. Just bear in mind I'm not that great with elisp. Also, sorry if this is just me failing to read some crucial piece of documentation.

**Comment by LuigiPiucco** [comment]
I managed to get it working. Turns out it's because I'm using DOOM Emacs, and they haven't yet updated to the commit that fixes #346. I unpinned the package to use master and removed the port business in the previous comment and it worked. I'm closing this issue and opening a new one over there to inform them of the update. Sorry for the noise...

**Comment by gagbo** [comment]
If that still works it should also probably be added in `dap-lldb.el`
